### ISSUE TITLE
Expressly allow <header> and <footer> to be scoped to <main> and not be landmarks

### ISSUE THREAD

**Issue by jasonkiss** [w3c/html#520]
See discussion at [ARIA issue #277](https://github.com/w3c/aria/issues/277). 

The TL;DR from that discussion:
- `header` or `footer` in a `main` element, but not otherwise descendant of a sectioning element, should not be scoped to the `body` element and should not be an implicit ARIA landmark.
- `header` is recommended as a [grouping mechanism for headings, subheadings, and taglines](http://w3c.github.io/html/common-idioms-without-dedicated-elements.html#subheadings-subtitles-alternative-titles-and-taglines)

The [definition of `header`](http://w3c.github.io/html/sections.html#the-header-element) says that "When the nearest ancestor sectioning content or sectioning root element is the `body` element, then it applies to the whole page." This makes the `header` an implicit `banner` landmark, but does not allow for the scenario where `header` is a descendant of `main`, but not a descendant of a sectioning element, and not intended to be a `banner` landmark.

Similarly, the [definition of `footer`](http://w3c.github.io/html/sections.html#the-footer-element) says that "When the nearest ancestor sectioning content or sectioning root element is the `body` element, then it applies to the whole page." This makes the `footer` an implicit `contentinfo` landmark, but does not allow for the scenario where `footer` is a descendant of `main`, but not a descendant of a sectioning element, and not intended to be a `contentinfo` landmark.

Recommend that the definitions of `header` and `footer` be updated to reflect that:
- they can represent introductory or footer content, respectively, for their ancestor `main` element, or nearest sectioning content or sectioning root element
- that when their nearest ancestor sectioning content or sectioning root element is the `body` element, then they apply to the whole page, unless they are descendants of `main`.
